pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -77,7 +77,9 @@ func (c *Cluster) makeDeployment(mgrConfig *mgrConfig) *apps.Deployment {
 		podSpec.Spec.DNSPolicy = v1.DNSClusterFirstWithHostNet
 	}
 	c.annotations.ApplyToObjectMeta(&podSpec.ObjectMeta)
-	c.applyPrometheusAnnotations(&podSpec.ObjectMeta)
+	if err := c.applyPrometheusAnnotations(&podSpec.ObjectMeta); err != nil {
```

Comment:
`applyPrometheusAnnotations()` always returns nil. Let's change that method to not return an error.